### ISSUE TITLE
Running a feature file containing unicode characters in examples causes Behave to exit with UnicodeEncodeError in some cases

### ISSUE THREAD

**Issue by kurtlinden** [behave/behave#805]
Using Behave 1.2.6

Running any feature file that contains example data containing unicode such as: アイウエオカキアイウエオカキアイイ
will cause Behave to exist with error:

UnicodeEncodeError: 'ascii' codec can't encode characters in position 0-16: ordinal not in range(128)

The stack trace shows this:

/usr/local/bin/behave in <module>()
      6 if __name__ == '__main__':
      7     sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
----> 8     sys.exit(main())

/usr/local/lib/python3.5/dist-packages/behave/__main__.py in main(args)
    181     """
    182     config = Configuration(args)
--> 183     return run_behave(config)
    184 
    185 if __name__ == "__main__":

/usr/local/lib/python3.5/dist-packages/behave/__main__.py in run_behave(config, runner_class)
    125         reset_runtime()
    126         runner = runner_class(config)
--> 127         failed = runner.run()
    128     except ParserError as e:
    129         print(u"ParserError: %s" % e)

/usr/local/lib/python3.5/dist-packages/behave/runner.py in run(self)
    802         with self.path_manager:
    803             self.setup_paths()
--> 804             return self.run_with_paths()
    805 
    806     def run_with_paths(self):

/usr/local/lib/python3.5/dist-packages/behave/runner.py in run_with_paths(self)
    822         stream_openers = self.config.outputs
    823         self.formatters = make_formatters(self.config, stream_openers)
--> 824         return self.run_model()

/usr/local/lib/python3.5/dist-packages/behave/runner.py in run_model(self, features)
    624                         formatter.uri(feature.filename)
    625 
--> 626                     failed = feature.run(self)
    627                     if failed:
    628                         failed_count += 1

/usr/local/lib/python3.5/dist-packages/behave/model.py in run(self, runner)
    319                     continue
    320 
--> 321                 failed = scenario.run(runner)
    322                 if failed:
    323                     failed_count += 1

/usr/local/lib/python3.5/dist-packages/behave/model.py in run(self, runner)
   1112         for scenario in self.scenarios:     # -- REQUIRE: BUILD-SCENARIOS
   1113             runner.context._set_root_attribute("active_outline", scenario._row)
-> 1114             failed = scenario.run(runner)
   1115             if failed:
   1116                 failed_count += 1

/usr/local/lib/python3.5/dist-packages/behave/model.py in run(self, runner)
    709             for step in self.all_steps:
    710                 if run_steps:
--> 711                     if not step.run(runner):
    712                         # -- CASE: Failed or undefined step
    713                         #    Optionally continue_after_failed_step if enabled.

/usr/local/lib/python3.5/dist-packages/behave/model.py in run(self, runner, quiet, capture)
   1309         if not quiet:
   1310             for formatter in runner.formatters:
-> 1311                 formatter.match(match)
   1312 
   1313         if capture:

/usr/local/lib/python3.5/dist-packages/behave/formatter/pretty.py in match(self, match)
    130         self.print_statement()
    131         self.print_step(Status.executing, self._match.arguments,
--> 132                         self._match.location, self.monochrome)
    133         self.stream.flush()
    134
/usr/local/lib/python3.5/dist-packages/behave/formatter/pretty.py in print_step(self, status, arguments, location, proceed)
    299             self.stream.write(text_format.text(text))
    300             line_length += len(text)
--> 301             self.stream.write(arg_format.text(arg.original))
    302             line_length += len(arg.original)
    303             text_start = arg.end

Note:  This did not occur with Python 2.7 but does occur with Python 3.5.

**Comment by jenisys** [comment]
* Which operating system are you using ?
* What encoding are you using in your feature files ?
* What encoding are you using in your shell ?
* Do you observe the behavior above only with the pretty formatter or also with the plain formatter ?

**Comment by kurtlinden** [comment]
- Operating system:  Debian
- Using UTF-8 encoding in feature files (at least I think, is there a way to specify an encoding for a feature file?)
- Using UTF-8 encoding in shell (issue occurs on both Mac and PC shells)
- I see this error when using both the pretty and plain formatters

Thank you for your time and consideration.

**Comment by Prasannajnaeyulu** [comment]
Hi Any progress on this issue. I am also facing the same issue. Any workaround?

**Comment by kurtlinden** [comment]
Have not seen any progress on this issue, our work around is to run behave with the -f progress option (not ideal but works to get around the error)

**Comment by Prasannajnaeyulu** [comment]
Thanks for the reply. I just resolved this problem. For me the problem is, I’m in Ubuntu Docker environment where LANG and LANGUAGE environment variables all set with value “POSIX”.
Even I set these with the values “en_US.UTF-8” and “en_US:en” using export I couldn’t get the problem solved. The root cause for me is lack of language-pack-en. After I installed the package “language-pack-en” using “apt-get -y install language-pack-en“ and then I set the environment variables  LANG and LANGUAGE as
$export LANG= en_US.UTF-8
$export LANGUAGE=en_US:en

Now it’s working fine for me. It’s all to deal with making your shell support the UTF-8.
